RadClass/RadClass.py

```diff
@@ -86,28 +92,49 @@ def queue_file(self):
 
         self.processor = ds.DataSet(self.labels)
         self.processor.init_database(self.filename, self.datapath)
-        # parameters for keeping track of progress through a file
 
+        # parameters for keeping track of progress through a file
         self.start_i = 0
         self.stop_i = len(self.processor.timestamps) - 1
 
         if self.start_time is not None:
+            # invalid user input
+            if self.start_time > self.processor.timestamps[-1]:
+                raise ValueError('User start time is larger than all timestamps.')
             timestamp = self.processor.timestamps[self.processor.timestamps >=
                                                   self.start_time][0]
+            # search for start or choose first timestamp
             self.start_i = max(np.searchsorted(self.processor.timestamps,
                                                timestamp,
                                                side='right')-1, 0)
```

Comment:
Sorry - new query/concern: What if you flipped these, and searched for the `start_time`, got the index, and then just pulled the timestamp out of the full list by its index?  It seems like this is effectively doing two searches.

Reply:
Agreed...now that I look at it, I'm not sure why the `np.searchsorted` is needed. You already know the timestamp, you just need its unique index. Let me tweak this.